### ISSUE TITLE
Accessibility issues found in DDL TestRail audit

### ISSUE THREAD

**Issue by jaredcunha** [department-of-veterans-affairs/vets-website#22576]
Documenting accessibility issues found in DDL as part of the test rails audit. (still ongoing - tagging @joshkimux for visibility)

 ## Dev work can be done immediately

- [x] #22632
- [x] Heading order is out of order. Jumps from `h1` to `h4`. Claims letter headings should be `h2` and sized with CSS Utility Classes from the design system
- [x] Use `<ol>` and `<li>` for the letters list.

## Needs content/design

- [x] #22634
- [x] #22633
- [ ] Page title needs to be changed. The page title is "Track Claims | Veterans Affairs" when it should be something like "Your VA Claim Letters | Veterans Affairs"
- [ ] Might be worth including in the description under the heading that letters are organized with the most recent first.
- [ ] Unique file names for letters to include date
